src/GitHub.VisualStudio/UI/Views/GitHubConnectContent.xaml.cs

```diff
@@ -39,23 +41,74 @@ void signOut_Click(object sender, RoutedEventArgs e)
             signOut.IsEnabled = true;
         }
 
-        private void login_Click(object sender, RoutedEventArgs e)
+        void login_Click(object sender, RoutedEventArgs e)
         {
             login.IsEnabled = false;
             ViewModel.Login();
             login.IsEnabled = true;
         }
 
-        public GitHubConnectSection ViewModel
+        void repositories_MouseDoubleClick(object sender, System.Windows.Input.MouseButtonEventArgs e)
         {
-            get { return (GitHubConnectSection)GetValue(ViewModelProperty); }
+            ViewModel.OpenRepository();
+            e.Handled = true;
+        }
+
+        public IGitHubConnectSection ViewModel
+        {
+            get { return (IGitHubConnectSection)GetValue(ViewModelProperty); }
             set { SetValue(ViewModelProperty, value); }
         }
 
         public static readonly DependencyProperty ViewModelProperty =
             DependencyProperty.Register(
                 "ViewModel",
-                typeof(GitHubConnectSection),
+                typeof(IGitHubConnectSection),
                 typeof(GitHubConnectContent));
     }
+
+    public class FormatRepositoryNameConverter : IMultiValueConverter
+    {
+        public object Convert(object[] values, Type targetType, object parameter, CultureInfo culture)
+        {
+            if (values.Length != 2 || parameter as string != "FormatRepositoryName")
+                return String.Empty;
+
+            if (!(values[1] is IGitHubConnectSection))
+            {
+                if (values[0] is ISimpleRepositoryModel)
+                    return ((ISimpleRepositoryModel)values[0]).Name;
+                return String.Empty;
```

Comment:
This is probably more a code style issue than an actual perf issue since array access and casting types is pretty cheap, but I would probably write the above as:

It probably doesn't matter for perf, but I tend to replace double cast checks with something like:

```csharp
var model = values[0] as ISimpleRepositoryModel;
return model != null ? model.Name : String.Empty;
```

That accesses the array only once and type checks only once. However, more importantly to me, I think that's a bit more readable.